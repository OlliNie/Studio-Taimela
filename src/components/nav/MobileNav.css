.nav{
  z-index: 99;
  background-color: rgba(0,0,0,.5);
  display:flex;
  flex-direction: column;
  position:fixed;
  width:100%;
  top:0px;
  @media (min-width: 720px){
   
    display: none;
  }
}
.navBar{
  margin:0;
  @media (min-width: 720px){
    margin: 0 4rem;
  }
  width:100%;
  display:flex;
 
  align-items:center;
  img{
    height: 5rem;
    width: 5rem;
    margin-right:auto;
    padding-left:2rem;
   }
  svg{
    padding-right:2rem;
  }
}
.navMenu{
  width:100%;
  display:flex;
  justify-content: center;
  border-top:1px solid black;
  font-size:1.5rem;
  ul{
    padding:0;
    width:100%;
    text-align: center;
  }
  a{
    color:inherit;
    text-decoration: none;
  }
  li{
    margin:1rem 0;
    width:100%;
  }
}