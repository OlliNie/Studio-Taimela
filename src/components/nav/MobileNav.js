import React, { PureComponent } from 'react';
import logo from '../../assets/images/logo.png';
import { FaBars } from 'react-icons/fa';
import styles from './MobileNav.css';
import createLinks from './createLinks';
import { allLinks } from '../../text/nav';


export default class MobileNav extends PureComponent {
  state = {
    clicked: false
  }
  onClick = () => {
    this.setState((state) => {
      return { clicked: !state.clicked };
    });
  }
  render() {
    const mobileList = createLinks(allLinks, this.onClick);
    return (
      <>
        <nav className={styles.nav}>
          <section className={styles.navBar}>
            <img alt='logo' src={logo}></img>
            <FaBars onClick={this.onClick} size={32} />
          </section>
        </nav>
        <section className={[styles.navMenu, this.state.clicked].join(' ')}>
          {this.state.clicked === true ? <ul>{mobileList}</ul> : null}
        </section>
      </>
    );
  }
}

