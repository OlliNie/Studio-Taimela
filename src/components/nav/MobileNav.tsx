import createLinks from './createLinks.js';
import { allLinks } from '../../text/nav.js';
import * as React from 'react';

const FunctionComponent: React.FC = () => {
  const [navState, setNavState] = React.useState( {
    display:false
  }) 
  const updateClick = ()=>{
    setNavState({display:!navState.display});
  };

  const mobileList = createLinks(allLinks, updateClick);

  return(<h1>hi</h1>)
}

export default FunctionComponent;