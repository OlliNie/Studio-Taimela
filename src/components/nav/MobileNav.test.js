import React from 'react';
import { shallow } from 'enzyme';
import MobileNav from './DesktopNav';

describe('MobileNav Component', () => {
  it('renders mobile nav', () => {
    const wrapper = shallow(<MobileNav />);
    expect(wrapper).toMatchSnapshot();
  });
});
