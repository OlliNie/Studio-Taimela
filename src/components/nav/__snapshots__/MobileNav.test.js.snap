// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav Component renders mobile nav 1`] = `
<div
  className="container"
  id="Nav Container"
>
  <nav
    className="nav"
  >
    <section
      className="leftLinkContainer"
    >
      <ul
        className="hover"
      >
        <li
          key="0"
        >
          <HashLink
            to="#Expertise"
          >
            Expertise
          </HashLink>
        </li>
        <li
          key="1"
        >
          <HashLink
            to="#Services"
          >
            Services
          </HashLink>
        </li>
      </ul>
    </section>
    <section
      className="logoContainer"
    >
      <img
        alt="logo"
        src="/path/image"
      />
    </section>
    <section
      className="rightLinkContainer"
    >
      <ul>
        <li
          key="0"
        >
          <HashLink
            to="#Background"
          >
            Background
          </HashLink>
        </li>
        <li
          key="1"
        >
          <HashLink
            to="#Contact"
          >
            Contact
          </HashLink>
        </li>
      </ul>
    </section>
  </nav>
</div>
`;
